metric-anomaly-detector/src/main/java/org/hypertrace/alert/engine/metric/anomaly/detector/MetricAnomalyDetector.java

```diff
@@ -102,14 +111,39 @@ void process(AlertTask alertTask) {
         Instant.ofEpochMilli(alertTask.getCurrentExecutionTime()));
 
     if (violationCondition.hasStaticThresholdCondition()) {
-      evaluateForStaticThreshold(
-          iterator,
-          violationCondition,
-          metricAnomalyEventCondition.getMetricSelection().getMetricAttribute());
+      EvaluationResult evaluationResult =
+          evaluateForStaticThreshold(
+              iterator,
+              violationCondition,
+              metricAnomalyEventCondition.getMetricSelection().getMetricAttribute());
+      if (evaluationResult.isViolation()) {
+
+        MetricAnomalyNotificationEvent metricAnomalyNotificationEvent =
+            MetricAnomalyNotificationEvent.newBuilder()
+                .setViolationTimestamp(alertTask.getCurrentExecutionTime())
+                .setChannelId(alertTask.getChannelId())
+                .setEventConditionId(alertTask.getEventConditionId())
+                .setEventConditionType(alertTask.getEventConditionType())
+                .build();
+        EventRecord eventRecord =
+            EventRecord.newBuilder()
+                .setEventType(METRIC_ANOMALY_ACTION_EVENT_TYPE)
+                .setEventRecordMetadata(Map.of())
+                .setEventValue(metricAnomalyNotificationEvent.toByteBuffer())
+                .build();
+        NotificationEvent notificationEvent =
+            NotificationEvent.newBuilder()
+                .setTenantId(alertTask.getTenantId())
+                .setActionEventMetadata(Map.of())
+                .setEventTimeMillis(alertTask.getCurrentExecutionTime())
+                .setEventRecord(eventRecord)
+                .build();
```

Comment:
Both setEventTimeMillis and setViolationTimestamp are same, shouldn't we store current time in setEventTimeMillis?

Reply:
Storing currentTime